models/Donation.js

```diff
@@ -0,0 +1,50 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const CustomerSchema = new Schema({
+  id: Number,
+  name: String,
+  phone_number: String,
+  email: String,
+  created_at: Date
+});
+
+const TransactionSchema = new Schema({
+  id: {
+    type: Number,
+    required: true,
+    unique: true
+  },
+  tx_ref: String,
+  flw_ref: String,
+  amount: Number,
+  currency: String,
+  charged_amount: Number,
+  app_fee: Number,
+  merchant_fee: Number,
+  processor_response: String,
+  auth_model: String,
+  status: String,
+  payment_type: String,
+  created_at: Date,
+  account_id: Number,
+  customer: {
+    type: CustomerSchema
+  }
+});
+
+// The main schema
+const DonationSchema = new Schema({
```

Comment:
Why do we need a new schema for this? Is it for the Flutterwave webhook?

Reply:
Yes, it's for the Flw webhook. The actual data is an object within the payload that gets sent back. I divided the file into smaller schemas for coherence.
I'll rename it, though.